### ISSUE TITLE
Error statement for actions rpcs

### ISSUE THREAD

**Issue by BalazsLengyel** [netmod-wg/yang-next#149]
In some cases for rpcs and actions there is a need to supply detailed error information beyond what is specified in RFC 6241. NETCONF allows for additional information in the "error-message" or "error-info" elements, however there is no way to define the structure and syntax of the error message. 
In RFC 8639 we see an example of how to specify the structure of error information: 
     rc:yang-data establish-subscription-stream-error-info) {...}
While this solves defining the structure, only the plain English description text connects this error structure to the specific rpc/action.

It is proposed  to add an error substatement  for action, rpcs beside the input and output parts.
The error statement may contain any substatement that a container can contain except notification, action, config 

An alternative would be to contain all the error information in the "output" section of the action, rpc, however this would result in an rpc-reply instead of an rpc-error message even if the action/rpc failed.

**Comment by janlindblad** [comment]
Good idea. Would simplify for both YANG module authors, protocol extenders and implementers.

**Comment by reshad-equinix** [comment]
Support.

**Comment by kwatsen** [comment]
Andy: Re: proto-binding document, CORECONF is severally limited in supporting "modify" (just create/delete)
